src/utils/filesystem.cpp

```diff
@@ -843,6 +844,43 @@ bool verify_file(const std::string &fname,
     return true;
 }
 
+bool verify_file_size(const std::string &fname, const int64_t &expected_fsize)
+{
+    if (!file_exists(fname)) {
+        derror_f("file({}) is not existed", fname);
+        return false;
+    }
+    int64_t f_size = 0;
+    if (!file_size(fname, f_size)) {
+        derror_f("verify file({}) size failed, becaused failed to get file size", fname);
+        return false;
+    }
+    if (f_size != expected_fsize) {
+        derror_f("verify file({}) size failed, because file damaged, size: {} VS {}",
+                 fname,
+                 f_size,
+                 expected_fsize);
+        return false;
+    }
+    return true;
+}
+
+bool verify_data(const std::string &fname,
+                 const char *data,
+                 const size_t data_size,
+                 const std::string &expected_md5)
+{
+    std::string md5 = string_md5(data, data_size);
```

Comment:
Had you check the `string_md5` result is equal with the origin `md5sum`?

Reply:
+1